account_payment_partner/models/account_move.py

```diff
@@ -56,7 +56,7 @@ def _compute_partner_bank_filter_type_domain(self):
     @api.depends("partner_id", "company_id")
     def _compute_payment_mode(self):
         for move in self:
-            move.payment_mode_id = False
+            move.payment_mode_id = move.payment_mode_id
```

Comment:
What happens if the company changes?

Reply:
I think that this change would not affect, but it is important that it is not set to False because if for example we want to define a default value we could not (with this change if it would be possible).

Reply:
You can add following change for getting the desired result on company change:
```suggestion
            if move.payment_mode_id.company_id == move.company_id:
                move.payment_mode_id = move.payment_mode_id
            else:
                move.payment_mode_id = False
```